rotkehlchen/api/rest.py

```diff
@@ -829,6 +859,15 @@ def query_blockchain_balances(
             status_code = HTTPStatus.BAD_GATEWAY
         else:
             result = balances.serialize()
+            # Filter balances by threshold after serialization
+            if usd_value_threshold is not None and result is not None:
+                # Filter per chain balances
+                for chain, chain_balances in result.items():
+                    result[chain] = {
+                        asset: balance for asset, balance in chain_balances.items()
+                        if 'usd_value' in balance and FVal(balance['usd_value']) >
```

Comment:
You don't need `'usd_value' in balance` since the serialization always adds it. Otherwise is a programming error but correct me if something is different here

```suggestion
                        if FVal(balance['usd_value']) > usd_value_threshold
```

Reply:
I just realized. Why not do this for loop before calling to serialize? You avoid serializing entries that you will discard later